content/docs/setup/kubernetes/additional-setup/cni/index.md

```diff
@@ -40,79 +40,19 @@ replaces the functionality provided by the `istio-init` container.
     *  The Kubernetes documentation highly recommends this for all Kubernetes installations
        where `ServiceAccounts` are utilized.
 
-1.  [Download the Istio release](/docs/setup/kubernetes/#downloading-the-release).
-
 ## Installation
 
 1.  Determine the Kubernetes environment's CNI plugin `--cni-bin-dir` and `--cni-conf-dir` settings.
+    Refer to [Hosted Kubernetes settings](#hosted-kubernetes-settings) for any non-default settings required.
 
-    1.  Refer to the [Hosted Kubernetes Usage](#hosted-kubernetes-usage) section for any non-default
-        settings required.
-
-1.  Create a namespace for the Istio control-plane components:
-
-    {{< text bash >}}
-    $ kubectl create namespace istio-system
-    {{< /text >}}
-
-1.  Expand the release's compressed Istio CNI chart.
-
-    {{< text bash >}}
-    $ tar -xzf install/kubernetes/helm/charts/istio-cni-*.tgz -C install/kubernetes/helm/
-    {{< /text >}}
-
-1.  Install Istio CNI.
-
-    *  Option 1.  Render and apply the Istio CNI Helm template.
-
-        {{< text bash >}}
-        $ helm template install/kubernetes/helm/istio-cni --name=istio-cni --namespace=istio-system | kubectl apply -f -
-        {{< /text >}}
-
-    *  Option 2.  Install via Helm and Tiller via `helm install`.
-
-        {{< text bash >}}
-        $ helm install install/kubernetes/helm/istio-cni --name=istio-cni --namespace=istio-system
-        {{< /text >}}
+1.  Install Istio CNI and Istio using Helm.
```

Comment:
Per my previous comments I am not sure why feel compelled to include the Istio install here.  The rest of this page is not dependent on any Istio installation steps so it isn't required to install the CNI.  If you few strongly that the Istio install should be left here at least change the wording to say something "it is also recommended to install Istio at this time".  Practically it would be better to not install Istio at the same time as any containers that require a sidecar are dependent on the CNI being up before they can commence.

Reply:
It _is_ intertwined with the Istio installation steps. Especially, the following steps must be completed beforehand: download the release, install Helm (incl. creating the account for Tiller), create the `istio-system` namespace.

Reply:
A couple points.   You don't have to download the release its just the most convenient way to get the helm chart.   You also don't need to install the CNI in the istio-system namespace it works just fine in its own namespace.  We also document an option that doesn't require Tiller.   So it is absolutely not required to install Istio to get the CNI installed.   If you want to keep it the Istio install in the documentation at least note it is recommended and not required.  Alternatively and slightly better would be to note why it is recommended to install Istio as you have noted in your response above (i.e. Chart, ns & accounts)    Also the implied order here is wrong as you only want to follow enough of the Istio install to get the helm chart, create the necessary accounts and create the namespace.   Then you want to complete the CNI install before completing the Istio install otherwise some of the Istio pods will not install until the CNI is up.

Reply:
@john-a-joyce  -- IMO, for the purposes of this doc the Istio and istio-cni installs are intertwined.  I went through the same discussion in release 1.1 about just describing purely the istio-cni install and the opinions were that it should be contextualized along with the getting started guide's procedure for Istio install.  There also was a strong push for documentation simplicity over detailed scenarios so a lot of my original explanations about the independence of the CNI install were removed during review.  

@rlenglet -- We should convey that the istio-cni install works with any of the Istio install profiles and explain the step in "default with CNI enabled" tab for installing the istio-cni is identical for all profiles.  Possibly we want to pull out the istio-cni apply into its own step in the procedure like it was prior but add it to the helm install doc like you've started with the new tab.

Reply:
@tiswanso - I recall the history - but at the same time the Istio and istio-cni where more interwound at that time now they are almost entirely separated.   Although i feel strongly about this I seem to be in the minority on this point.   If the Istio install is left in these instructions then there MUST be some type of note to indicate that the CNI install should be completed before the bulk of the Istio install should be attempted.

Reply:
I still disagree that the istio install should be mention here at all.  As currently written it goes in circles.  
Istio Install docs - "go install CNI"
CNI install instructions - "go install Istio"

That is plain stupid in my opinion,  I don't really care what history go us to this point.  It is still stupid and confusing.  I am tired of arguing about it so this will be my last comment on the topic.  If people insist that Istio install should be included here in the CNI install steps I implore you to at least make sure the reader understands that the CNI should be installed first.

Reply:
@rlenglet -- I spoke with @john-a-joyce.  His concerns would be satisfied with a warning or blurb in the CNI installation doc to the effect of:
"The Istio installation may fail if the Istio CNI plugin is not installed prior."

Reply:
Sorry, catching up on this thread.

> You don't have to download the release its just the most convenient way to get the helm chart.

More than "the most convenient way", my view point is that this doc should specify **the recommended way** to install the Istio CNI plugin, which guarantees that it it will work regardless of how users decide to install Istio. If users decide to install Istio using Tiller, which is one of the recommended ways to install Istio, the doc should make it easy to work for them.

> We also document an option that doesn't require Tiller.

True, but we should document precisely how the Istio CNI needs to be installed in the Tiller case, since Tiller is a recommended way to install Istio, and installing Istio CNI in this scenario is far from trivial.

> Then you want to complete the CNI install before completing the Istio install otherwise some of the Istio pods will not install until the CNI is up.

How could that happen if the user followed the new install doc step-by-step?

> explain the step in "default with CNI enabled" tab for installing the istio-cni is identical for all profiles.

OK, I'll do that.

> Possibly we want to pull out the istio-cni apply into its own step in the procedure like it was prior but add it to the helm install doc like you've started with the new tab.

I decided against doing that because the installation of the `istio-cni` chart is strongly coupled with passing `--set istio_cni.enabled=true`. Putting those commands together highlights the strong dependency.

Reply:
@rlenglet I understand what you are trying to do is supply information that works for as many readers as possible. I understand that you want CNI to work for readers that are using Tiller and those that aren't, for example. Istio offers configuration profiles for installation. My advice is that you pick a configuration profile during install and write the document based on that configuration profile. Then you can just introduce this doc by saying in the Before you begin section: "Install Istio using [the configuration profile you select]."

Reply:
FWIW you can't install Istio first it will fail or at least spin its wheels until the CNI comes up.  By install I mean the actual kube apply of the Istio YAML some of the Istio install steps must be done first (e.g to download the release.